test/test_slacko.ml

```diff
@@ -0,0 +1,174 @@
+open Lwt
+open OUnit2
+
+open Slounit
+open Abbrtypes
+
+
+let token_str =
+  try Sys.getenv "SLACKO_TEST_TOKEN" with Not_found -> "xoxp-testtoken"
+
+let token = Slacko.token_of_string token_str
+let badtoken = Slacko.token_of_string "badtoken"
+
+(* If we have a non-default token, assume we want to talk to real slack. If
+   not, use our local fake instead. *)
+let () = match token_str with
+  | "xoxp-testtoken" -> Slacko.set_base_url "http://127.0.0.1:7357/api/"
+  | _ ->
+    print_endline ("NOTE: Because an API token has been provided, " ^
+                   "tests will run against the real slack API.");
+    try
+      (* We may want to talk to a proxy or a different fake slack. *)
+      let base_url = Sys.getenv "SLACKO_TEST_BASE_URL" in
+      Slacko.set_base_url base_url;
+      print_endline @@ "NOTE: Overriding slack base URL to " ^ base_url
+    with Not_found -> ()
+
+
+let abbr_json abbr_of_yojson json =
+  match abbr_of_yojson json with
+  | Ok abbr -> abbr
+  | Error err -> failwith @@ "Error parsing JSON: " ^ err
+
+let get_success = function
+  | `Success obj -> obj
+  | _ -> assert_failure "Unexpected failure."
+
+
+(* api_test *)
+
+let test_api_test_nodata tctx =
+  Slacko.api_test () >|= get_success >|= fun json ->
+  assert_equal ~printer:Yojson.Safe.to_string
```

Comment:
I like how it uses an already provided printer. Printing things can be so painful sometimes.